signal/src/lib.rs

```diff
@@ -558,11 +556,11 @@ pub mod pallet {
 
 			ProposalVoting::<T>::insert(&proposal_id, &voting);
 
-			// SBP-M2 review: what if proposal is not finalized?
-			// Is it properly handled?
 			// For Absolute majority if more then 50% of members vote for one option, the proposal period ends earlier.
 			if let Some(final_proposal_state) = Self::try_finalize_proposal(&voting) {
-				Self::finalize_proposal(&proposal_id, final_proposal_state, &voting);
+				Self::finalize_proposal(&proposal_id, final_proposal_state, &voting)?;
+			} else {
+				log::error!("No proposal state has been set for proposal_id: {:?}.", proposal_id);
```

Comment:
I can see many other cases why `finalize_proposal` could fail. Why the error here is only about the proposal state?